File: NoteBase_csharp/Core/NoteStatus.cs
namespace NoteBase.Core
{
    /// <summary>
    /// ノート単位のステータス。
    /// </summary>
    public enum NoteStatus
    {
        Active,
        Done,
        Pending,
        Archived,
    }

    public static class NoteStatusExtensions
    {
        public static string ToWireString(this NoteStatus s)
        {
            switch (s)
            {
                case NoteStatus.Active: return "active";
                case NoteStatus.Done: return "done";
                case NoteStatus.Pending: return "pending";
                case NoteStatus.Archived: return "archived";
                default: return "active";
            }
        }

        public static NoteStatus? ParseStatus(string s)
        {
            if (s == null) return null;
            switch (s)
            {
                case "active": return NoteStatus.Active;
                case "done": return NoteStatus.Done;
                case "pending": return NoteStatus.Pending;
                case "archived": return NoteStatus.Archived;
                default: return null;
            }
        }
    }
}

File: notebase/core/note_status.py
from __future__ import annotations

from enum import Enum


class NoteStatus(Enum):
    ACTIVE = "active"
    DONE = "done"
    PENDING = "pending"
    ARCHIVED = "archived"

    def to_wire(self) -> str:
        return self.value

    @classmethod
    def parse(cls, s: str | None) -> "NoteStatus | None":
        if s is None:
            return None
        for st in cls:
            if st.value == s:
                return st
        return None
